pyro/poutine/trace_struct.py

```diff
@@ -148,14 +151,21 @@ def compute_log_prob(self, site_filter=lambda name, site: True):
                 try:
```

Comment:
The traceback with the reraise seems a bit misleading for Python 3.

@fritzo: What do you think about changing this `try..except KeyError` block to checking for `if "log_prob" not in site`? My understanding is that `try...except` is more performant in a cache setting, but it might just be a blip given our dictionary size and access patterns. What do you think?

Reply:
It should be fine to replace it. The try-vs-if difference is negligible since the computation is expensive and we only call this function once or twice per trace.

Reply:
OK, I will do the change 😄

Reply:
Please take care to make the change in all `Trace.*log_prob*()` methods, so their implementations are still close.